### PR TITLE
Add must-gather test suite, add pod logs to artifacts dir on failure.

### PR DESCRIPTION
**Pull request by kaovilai**

### CODE REVIEW

**Comment by kaovilai** [comment]
test pr comment from gh cli

File: tests/e2e/backup_restore_suite_test.go [hunk 1]
```diff
@@ -351,14 +360,6 @@ var _ = Describe("AWS backup restore tests", func() {
...
 			Expect(err).ToNot(HaveOccurred())
 
-			// Test is successful, clean up everything
-			log.Printf("Uninstalling application for case %s", brCase.Name)
-			err = UninstallApplication(dpaCR.Client, brCase.ApplicationTemplate)
-			Expect(err).ToNot(HaveOccurred())
-
-			// Wait for namespace to be deleted
-			Eventually(IsNamespaceDeleted(brCase.ApplicationNamespace), timeoutMultiplier*time.Minute*2, time.Second*5).Should(BeTrue())
-
```
Comment (kaovilai) [inline tests/e2e/backup_restore_suite_test.go:361]: Moved to reportAfterEach

**Comment by kaovilai** [comment]
TODO: Must-gather test suite

**Review by sseago — APPROVED** [review]

File: tests/e2e/must-gather_suite_test.go [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+			ApplicationTemplate:  "./sample-applications/mongo-persistent/mongo-persistent-csi.yaml",
+			ApplicationNamespace: "mongo-persistent",
+			Name:                 "mongo-datamover-e2e",
+			BackupRestoreType:    CSIDataMover,
+			PreBackupVerify:      dataMoverReady(true, false, mongoready),
+			PostRestoreVerify:    dataMoverReady(false, false, mongoready),
+			MustGatherFiles: 	[]string{
+				"namespaces/openshift-adp/oadp.openshift.io/dpa-ts-example-velero/ts-example-velero.yml",
+				"namespaces/openshift-adp/velero.io/backupstoragelocations.velero.io/ts-example-velero-1.yaml",
+			},
```
Comment (kaovilai) [inline tests/e2e/must-gather_suite_test.go:367]: New file assertions.

**Review by mrnold — APPROVED** [review]

**Pull request merged**